### PR TITLE
Add validation for credential secret

### PR DESCRIPTION
**Pull request by FrankYang0529**
Issue: https://github.com/longhorn/longhorn/issues/811

### CODE REVIEW

File: datastore/longhorn.go [hunk 1]
```diff
@@ -107,6 +107,23 @@ func (s *DataStore) ValidateSetting(name, value string) (err error) {
 			}
 			return fmt.Errorf("cannot modify BackupTarget since there are existing standby volumes: %v", standbyVolumeNames)
 		}
+	case types.SettingNameBackupTargetCredentialSecret:
+		secret, err := s.kubeClient.CoreV1().Secrets(s.namespace).Get(value, metav1.GetOptions{})
```
Comment (shuo-wu) [inline datastore/longhorn.go:111]:
```suggestion
secret, err := s.GetSecretRO(s.namespace, value)
```

**Review by shuo-wu — COMMENTED** [review]
Thanks for your contribution! I think we can add the validation for other fields as well. e.g. `AWS_CERT`

https://github.com/longhorn/longhorn-manager/blob/d8b8bc23ca76da647f93e4e556f7b906ce8664a6/types/types.go#L127-L138

File: datastore/longhorn.go [hunk 2]
```diff
@@ -107,6 +107,23 @@ func (s *DataStore) ValidateSetting(name, value string) (err error) {
...
+		if err != nil {
+			return err
```
Comment (PhanLe1010) [inline datastore/longhorn.go:113]: return errors.Wrapf(err, "failed to list the secret %v before modifying backup target credential secret setting", value)

**Review by PhanLe1010 — CHANGES_REQUESTED** [review]
We we also change the defer statement to:
```
	defer func() {
		err = errors.Wrapf(err, "fail to set the setting %v with invalid value %v", name, value)
	}()
```

File: datastore/longhorn.go (hunk 1 above)
Comment (FrankYang0529) [reply datastore/longhorn.go]: Fixed it. Thank you.

File: datastore/longhorn.go (hunk 2 above)
Comment (FrankYang0529) [reply datastore/longhorn.go]: Fixed it. Thank you.

**Comment by shuo-wu** [comment]
@innobead I think we need to help backport this to v1.1.3 & 1.2.x

**Comment by innobead** [comment]
> @innobead I think we need to help backport this to v1.1.3 & 1.2.x

YEP, I think we should for bug fixes. cc @jenting

**Pull request merged**